functions/_fzf_preview_changed_file.fish

```diff
@@ -0,0 +1,24 @@
+function _fzf_preview_changed_file
+    set -l path (string split ' ' $argv)[-1]
+    if string match -r '^\?\?' $argv >/dev/null
+        echo Untracked
+        echo
+        _fzf_preview_file $path
+    else if string match -r '^A\s\s\S' $argv >/dev/null
+        echo -e (set_color green)New file
+        echo
+        git diff --color=always -- /dev/null $path
+    else
+        if string match -r '\S\s\S' $argv >/dev/null
+            echo -e (set_color red)Unstaged
+            echo
+            git diff --color=always -- $path
+            echo
+        end
+        if string match -r '\S\s\s\S|\S\S\s\S' $argv >/dev/null
```

Comment:
sorry I know you tried to explain what you were doing but I'm still confused about this regex and the one above

Reply:
Maybe `^\S` also works.